### ISSUE TITLE
[5.0.0rc13] application crash using OMPI with UCX

### ISSUE THREAD

**Issue by david-edwards-linaro** [open-mpi/ompi#11974]
Thank you for taking the time to submit an issue!

### What version of Open MPI are you using? (e.g., v3.0.5, v4.0.2, git branch name and hash, etc.)
5.0.0rc13 with UCX 1.15.0

### Describe how Open MPI was installed (e.g., from a source/distribution tarball, from a git clone, from an operating system distribution package, etc.)
tarball, using configure --with-ucx=ucx-1.15.0 --enable-oshmem

### Please describe the system on which you are running

* Operating system/version: Ubuntu 22.04
* Computer hardware: Intel(R) Core(TM) i7-1260P

-----------------------------

## Details of the problem

Please describe, in detail, the problem that you are having, including the behavior you expect to see, the actual behavior that you are seeing, steps to reproduce the problem, etc.  It is most helpful if you can attach a small program that a developer can use to reproduce your problem.

The following application crashes when built & run using Open MPI as configured above (it does not crash when OMPI is built without UCX):
```
#include "mpi.h"

int main(int argc, char *argv[])
{
    MPI_Init(&argc, &argv);

    MPI_Win win;
    MPI_Aint lowerbound;
    MPI_Aint sizeofreal;
    MPI_Type_get_extent(MPI_REAL, &lowerbound, &sizeofreal);
    float a[1] = {0};
    MPI_Win_create(a, sizeofreal, sizeofreal, MPI_INFO_NULL, MPI_COMM_WORLD, &win);
    MPI_Win_free(&win);

    MPI_Finalize();

    return 0;
}
```

The crash occurs within opal_common_ucx_support_level() due to dereference of a null pointer (*opal_common_ucx.tls).
The crash also occurs with 5.0.0rc13 using UCX 1.14.0.
The crash does not occur with 5.0.0rc12 using UCX 1.14.0.

**Comment by roiedanino** [comment]
@david-edwards-linaro 
Hi, can you please post here the crash output?

**Comment by david-edwards-linaro** [comment]
The stack trace is
```
(gdb) where
#0  __strcmp_avx2 () at ../sysdeps/x86_64/multiarch/strcmp-avx2.S:115
#1  0x00007ffff79369c6 in opal_common_ucx_support_level (context=0x5b0230) at ../../../../../openmpi-5.0.0rc13/opal/mca/common/ucx/common_ucx.c:248
#2  0x00007ffff7d41e7a in component_set_priority () at ../../../../../openmpi-5.0.0rc13/ompi/mca/osc/ucx/osc_ucx_component.c:312
#3  0x00007ffff7d42135 in component_query (win=0x5ad2f0, base=0x7fffffffd898, size=4, disp_unit=4, comm=0x404360 <ompi_mpi_comm_world>, info=0x560b20, flavor=1)
    at ../../../../../openmpi-5.0.0rc13/ompi/mca/osc/ucx/osc_ucx_component.c:367
#4  0x00007ffff7d36224 in ompi_osc_base_select (win=0x5ad2f0, base=0x7fffffffd898, size=4, disp_unit=4, comm=0x404360 <ompi_mpi_comm_world>, flavor=1, model=0x7fffffffd8ac)
    at ../../../../openmpi-5.0.0rc13/ompi/mca/osc/base/osc_base_init.c:57
#5  0x00007ffff7abb46c in ompi_win_create (base=0x7fffffffd954, size=4, disp_unit=4, comm=0x404360 <ompi_mpi_comm_world>, info=0x404060 <ompi_mpi_info_null>, newwin=0x7fffffffd968)
    at ../../openmpi-5.0.0rc13/ompi/win/win.c:257
#6  0x00007ffff7b567b9 in PMPI_Win_create (base=0x7fffffffd954, size=4, disp_unit=4, info=0x404060 <ompi_mpi_info_null>, comm=0x404360 <ompi_mpi_comm_world>, win=0x7fffffffd968)
    at ../../../../openmpi-5.0.0rc13/ompi/mpi/c/win_create.c:79
#7  0x00000000004011d0 in main (argc=1, argv=0x7fffffffda88) at test.c:12
```

**Comment by janjust** [comment]
fixed with #11985
